interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,301 @@
+    member="PauseOtherSessions"
+	peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Allow clients to query service properties
+dbus (send)
+    bus=system
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member="{Get,Set}"
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+   bus=system
+   path=/core/ubuntu/media/Service
+   interface=org.freedesktop.DBus.Properties
+   member=PropertiesChanged
+   peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.ObjectManager
+    peer=(label=unconfined),
+`)
+
+var mediaHubPermanentSlotSecComp = []byte(`
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var mediaHubConnectedPlugSecComp = []byte(`
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var mediaHubPermanentSlotDBus = []byte(`
+<policy user="root">
+    <allow own="core.ubuntu.media.Service"/>
+    <allow send_destination="core.ubuntu.media.Service"/>
+    <allow send_interface="core.ubuntu.media.Service"/>
+</policy>
+`)
+
+var mediaHubConnectedPlugDBus = []byte(`
+<policy context="default">
+    <deny own="core.ubuntu.media.Service"/>
+    <allow send_destination="core.ubuntu.media.Service"/>
+    <allow send_interface="core.ubuntu.media.Service"/>
+</policy>
+`)
+
+type MediaHubInterface struct{}
+
+func (iface *MediaHubInterface) Name() string {
+	return "media-hub"
+}
+
+func (iface *MediaHubInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		new := slotAppLabelExpr(slot)
+		snippet := bytes.Replace(mediaHubConnectedPlugAppArmor, old, new, -1)
+		return snippet, nil
+	case interfaces.SecurityDBus:
+		return mediaHubConnectedPlugDBus, nil
+	case interfaces.SecuritySecComp:
+		return mediaHubConnectedPlugSecComp, nil
+	}
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return mediaHubPermanentSlotAppArmor, nil
+	case interfaces.SecurityDBus:
+		return mediaHubPermanentSlotDBus, nil
+	case interfaces.SecuritySecComp:
+		return mediaHubPermanentSlotSecComp, nil
+	}
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		old := []byte("###PLUG_SECURITY_TAGS###")
+		new := plugAppLabelExpr(plug)
+		snippet := bytes.Replace(mediaHubConnectedSlotAppArmor, old, new, -1)
+		return snippet, nil
+	}
+	return nil, nil
```

Comment:
As mentioned above, please drop this and just put the mediaHubConnectedSlotAppArmor policy in mediaHubPermanentSlotAppArmor.